### PR TITLE
ci(circleci): try to simplify test matrix

### PR DESCRIPTION
**Pull request by lahabana**
Signed-off-by: Charly Molter <charly.molter@konghq.com>

### Summary

SUMMARY_GOES_HERE

### Full changelog

* [Implement ...]
* [Fix ...]

### Issues resolved

Fix #XXX

### Documentation

- [ ] Link to the website [documentation PR](https://github.com/kumahq/kuma-website/pull/XXX)

### Testing

- [ ] Unit tests
- [ ] E2E tests
- [ ] Manual testing on Universal
- [ ] Manual testing on Kubernetes

### Backwards compatibility

- [ ] Update [`UPGRADE.md`](/UPGRADE.md) with any steps users will need to take when upgrading.
- [ ] Add `backport-to-stable` label if the code follows our [backporting policy](/CONTRIBUTING.md#backporting)

### CODE REVIEW

**Comment by lahabana** [comment]
@jakubdyszkiewicz I think you the better suited person for this

File: .circleci/config.yml [hunk 1]
```diff
@@ -263,75 +246,90 @@ jobs:
...
+              parallelism:<< parameters.parallelism >> \
+            "
+
+            echo "some combinations may run nothing"
+      - unless:
+          # This works around circleci limitation by skipping tests for combinations that don't make sense
+          # See: https://discuss.circleci.com/t/matrix-exclude-entire-subset/43879
+          condition:
+            or:
+              - and:
```
Comment (jakubdyszkiewicz) [inline .circleci/config.yml:266]: could you comment on what is going on in this `and` branch?

File: .circleci/config.yml [hunk 2]
```diff
@@ -263,75 +246,90 @@ jobs:
...
+                      - not: {equal: [master, << pipeline.git.branch >>]}
+                  - or:
+                      - << parameters.ipv6 >>
+                      - equal: ["v1.19.16-k3s1", << parameters.k8sVersion >>]
+              - and:
+                # Ipv6 only works with kind
+                - << parameters.ipv6 >>
+                - not: {equal: [kind, << parameters.k8sVersion >>]}
+              - and:
+                # kind should only run with v6 except with universal
```
Comment (jakubdyszkiewicz) [inline .circleci/config.yml:278]: do we need this `and` and the previous `and`?

**Review by jakubdyszkiewicz — COMMENTED** [review]
I find `unless` section a bit hard to follow, but I'm ok with adopting this way of picking the job.

File: .circleci/config.yml (hunk 2 above)
Comment (lahabana) [reply .circleci/config.yml:278]: Yes it's an or. The first says `if ipv6 and not kind` don't run the other one says: `if not ipv6 and kind and not universal test`. Which is possible, I had a version with a `none` version but I found it even more complicated.

**Comment by lahabana** [comment]
@jakubdyszkiewicz review please :)

**Review by jakubdyszkiewicz — APPROVED** [review]
